clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/ServerGroupHandler.java

```diff
@@ -20,7 +20,7 @@
 import com.netflix.spinnaker.clouddriver.kubernetes.caching.view.model.KubernetesServerGroup;
 import com.netflix.spinnaker.clouddriver.kubernetes.caching.view.provider.data.KubernetesServerGroupCacheData;
 
-public interface ServerGroupHandler {
+public interface ServerGroupHandler extends CanReceiveTraffic {
```

Comment:
You need to also add this interface to `ServerGroupManagerHandler`, that's the handler for kubernetes `Deployment` objects. Probably it's not actually used for red/black deployments but to keep semantics correct, `Deployment` can also receive traffic.